### DISCUSSION TITLE
Ability to Midi map everything

### DISCUSSION THREAD

**Discussion by Ponchomouse** [SynthstromAudible/DelugeFirmware#2210] [Ideas]
Great work all, the midi follow feature in 1.1 is amazing.made me think how good it would be to be able to midi map all controls. One key one is the fill (sync scale) button as that is getting a lot more use than it was designed for. (and thanks for fill mode)

**Comment by Dust-r** [comment]
I want this, too.

**Comment by seangoodvibes** [comment]
> Great work all, the midi follow feature in 1.1 is amazing.made me think how good it would be to be able to midi map all controls. One key one is the fill (sync scale) button as that is getting a lot more use than it was designed for. (and thanks for fill mode) 

Re: Fill, there is a global midi command for fill that you can learn :)

**Comment by bukru** [comment]
Saturation too :p

**Comment by hsensh** [comment]
I would love to MIDI map something like LFO synced rate (8th note and 16 note and all), maybe we can have either CC being distributed among the possible values in equal amounts (like if there are 8 values we can split 255 by 8 and then make these continuous range belong to one discreet value at a time)

We could also have each discreet value be activated by a MIDI toggle. I find myself limited with my songs because some values I would like to MIDI map are not possible to map because they're discreet values rather than continuous.

**Comment by ScottHsuMusic** [comment]
Would love to MIDI map to the <> + Play feature. My performance workflow involves recording guitar loops in song mode, having them mapped into arranger mode, and then pressing <> + Play to initiate the start of arranger mode, but this requires me to use both hands which can be awkward. Would be great to have this mappable in the MIDI>CMD menu!

**Reply by seangoodvibes** [reply]
> Would love to MIDI map to the <> + Play feature. My performance workflow involves recording guitar loops in song mode, having them mapped into arranger mode, and then pressing <> + Play to initiate the start of arranger mode, but this requires me to use both hands which can be awkward. Would be great to have this mappable in the MIDI>CMD menu!

You could also scroll to beginning of arranger and just press play

**Reply by ScottHsuMusic** [reply]
> > Would love to MIDI map to the <> + Play feature. My performance workflow involves recording guitar loops in song mode, having them mapped into arranger mode, and then pressing <> + Play to initiate the start of arranger mode, but this requires me to use both hands which can be awkward. Would be great to have this mappable in the MIDI>CMD menu!
> 
> You could also scroll to beginning of arranger and just press play

Pressing play while playing in song mode would cause the song to stop. I'm looking to have the song play seamlessly into arranger mode when I'm performing live on stage.

**Reply by seangoodvibes** [reply]
> > > Would love to MIDI map to the <> + Play feature. My performance workflow involves recording guitar loops in song mode, having them mapped into arranger mode, and then pressing <> + Play to initiate the start of arranger mode, but this requires me to use both hands which can be awkward. Would be great to have this mappable in the MIDI>CMD menu!
> 
> > 
> 
> > You could also scroll to beginning of arranger and just press play
> 
> 
> 
> Pressing play while playing in song mode would cause the song to stop. I'm looking to have the song play seamlessly into arranger mode when I'm performing live on stage.

Ah

Have you seen the global midi command for restarting playback? We should already have that, I saw it in the code and it's in the manual

It's always been there

**Reply by seangoodvibes** [reply]
> > > Would love to MIDI map to the <> + Play feature. My performance workflow involves recording guitar loops in song mode, having them mapped into arranger mode, and then pressing <> + Play to initiate the start of arranger mode, but this requires me to use both hands which can be awkward. Would be great to have this mappable in the MIDI>CMD menu!
> 
> > 
> 
> > You could also scroll to beginning of arranger and just press play
> 
> 
> 
> Pressing play while playing in song mode would cause the song to stop. I'm looking to have the song play seamlessly into arranger mode when I'm performing live on stage.

![image](https://github.com/user-attachments/assets/4f526ba0-53bd-46a6-b22e-49a1352d0a4f)

See "Restart"

**Reply by ScottHsuMusic** [reply]
Thanks for the suggestion, unfortunately it's not exactly what I'm looking for. Restart immediately begins playback so I'd have to immediately start playing/singing after triggering. I'm looking for a map for <> + PLAY because it queues the beginning of Arranger Mode after the current loop finishes.
